hibernate-core/src/main/java/org/hibernate/resource/beans/container/spi/BeanContainer.java

```diff
@@ -27,9 +27,38 @@ <B> ContainedBean<B> getBean(
 			LifecycleOptions lifecycleOptions,
 			BeanInstanceProducer fallbackProducer);
 
+	/**
+	 * Form of {@link #getBean(Class, LifecycleOptions, BeanInstanceProducer)}
+	 * allowing to indicate whether it is required to use CDI if it is available
+	 *
+	 * @implNote Defaulted for backwards compatibility
+	 */
+	default <B> ContainedBean<B> getBean(
+			Class<B> beanType,
+			LifecycleOptions lifecycleOptions,
+			BeanInstanceProducer fallbackProducer,
+			boolean cdiRequiredIfAvailable) {
```

Comment:
Maybe another name would be better, since `BeanContainer` can be used for non-CDI implementations? E.g. `beanContainerRequiredIfAvailable` or something like that.